mobile/src/libs/hooks/use-search/use-search.hook.ts

```diff
@@ -0,0 +1,35 @@
+import { useAppForm, useCallback, useMemo } from '#libs/hooks/hooks';
+
+import { DEFAULT_SEARCH_PAYLOAD } from './libs/constants';
+
+type UseSearchResult<T> = {
+  filteredData: T[];
+  setSearchQuery: (value: string) => void;
+};
+
+const useSearch = <T>(data: T[], propertyName: keyof T): UseSearchResult<T> => {
+  const { setValue, watch } = useAppForm({
+    defaultValues: DEFAULT_SEARCH_PAYLOAD,
+  });
+
+  const setSearchQuery = useCallback(
+    (searchValue: string) => {
+      setValue('search', searchValue);
+    },
+    [setValue],
+  );
```

Comment:
I suppose it is useless because the control of input search should handle it

Reply:
Save data to state and check on watch of 'search' when should useEffect trigger setState

Reply:
Sorry, could you please clarify this for me? If I remove useCallback - setSearchQuery triggers many times. Do I have to handle this behaviour inside input-search, not inside use-search-hook?
And about data - do you mean filteredData?